app-games/src/main/java/com/aptoide/android/aptoidegames/gamegenie/presentation/composables/companion/ChatScreenCompanion.kt

```diff
@@ -28,48 +95,261 @@ fun ChatScreenCompanion(
   uiState: GameGenieUIState,
   navigateTo: (String) -> Unit,
   setFirstLoadDone: () -> Unit,
-  onMessageSend: (String) -> Unit,
+  onMessageSend: (String, String?) -> Unit,
   isLoading: Boolean = false,
+  showBottomSheet: ((BottomSheetContent?) -> Unit)? = null,
+  analytics: GameGenieAnalytics,
 ) {
-  Column(
-    modifier = Modifier
-      .padding(horizontal = 18.dp)
-      .fillMaxSize()
-  ) {
-    val showImage = uiState.chat.conversation.size < 2
+  val context = LocalContext.current
```

Comment:
You should have a repository (that usualy for shared preferences, we call XXXLocalRepository) to get these preferences. You have some examples on the code. You could either create an object with all the info you need and return it at once, or get it one by one, depending on how they are necessary